### PR TITLE
Stats: Now only valid probabilities can be entered in Bernoulli, Coin and FiniteRV

### LINKED ISSUE #15903: FiniteRV allows non valid probability distribution
https://github.com/sympy/sympy/blob/master/sympy/stats/frv_types.py#FiniteRV allows non valid probability distribution. It allows total probability not equal to 1. And also individual events' probabilities >1 or <0 are also allowed.
To reproduce the issue 

    >>>from sympy.stats import FiniteRV, P, E
    >>> density = {0: .1, 1: .2, 2: .3}
    >>> X = FiniteRV('X', density)

This will work. You can check by using `E(X)` or `P(X)` that it works even if it wrong.
Other examples are 
` >>> density = {0: .1, 1: .2, 2: .3, 3: 1.4} `
or
` >>> density = {0: .1, 1: .2, 2: .3, 3: -1.4} `
I think the whole https://github.com/sympy/sympy/blob/master/sympy/stats/frv_types.py may have this kind of problem.

### PR DESCRIPTION
**Pull request by supreet11agrawal**
#### References to other Issues or PRs

Fixes #15903 

#### Brief description of what is fixed or changed
frv_types earlier allowed non-valid probability distribution in case of Bernoulli, Coin and FiniteRV methods. After the modification, only probabilites with value in between 0 to 1 can be entered. Also in FiniteRV, earlier a random variable with non valid probability density was allowed. New changes prohibit such shortcomings.

Example:

Bernoulli
```
>>> from sympy.stats import Bernoulli, density
>>> from sympy import S
>>> X = Bernoulli('X', S(5)/4) #probability can't be >1 but 5/4 here
```
 Earlier this worked

Now it gives
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy/stats/frv_types.py", line 200, in Bernoulli
    return rv(name, BernoulliDistribution, p, succ, fail)
  File "sympy/stats/frv_types.py", line 28, in rv
    density = cls(*args)
  File "sympy/stats/frv_types.py", line 172, in __new__
    raise ValueError("'p' must be: 0 <= p <= 1 . p = %s" % str(p))
ValueError: 'p' must be: 0 <= p <= 1 . p = 5/4
```
Similarily, Coin and FiniteRV modified. Can be tested using invalid probability distribution

For FiniteRV, density's each element should be between 0 and 1 and also total density should be 1. This is now implemented.

#### Release Notes

*  stats
    * fixed bugs in finite discrete random variable types

### CODE REVIEW

**Comment by supreet11agrawal** [comment]
@Upabjojr  @jksuom Can you please review?

**Comment by jksuom** [comment]
There are some repeated imports, otherwise this looks good. There could also be a test for the issue.

**Comment by supreet11agrawal** [comment]
Which imports are you referring to? I don't see any repeated ones
I will add tests as well
Thanks +1:

**Comment by jksuom** [comment]
This is from frv_types.py:
```
from sympy import (S, sympify, Rational, binomial, cacheit, Integer,
        Dict, Basic, KroneckerDelta, Dummy)
from sympy.concrete.summations import Sum
from sympy.core.compatibility import as_int, range
from sympy.core.logic import fuzzy_not, fuzzy_and
from sympy.stats.frv import (SingleFinitePSpace, SingleFiniteDistribution)
from sympy.concrete.summations import Sum
from sympy import (S, sympify, Rational, binomial, cacheit, Integer,
        Dict, Basic, KroneckerDelta, Dummy)
```

**Comment by supreet11agrawal** [comment]
Ohkk! Sorry, don't know how I missed that

**Comment by supreet11agrawal** [comment]
@jksuom Is that okay?
I will add tests in a different PR

**Comment by supreet11agrawal** [comment]
Mostly by tomorrow

File: sympy/stats/frv_types.py [hunk 1]
```diff
@@ -167,6 +174,15 @@ def Die(name, sides=6):
 class BernoulliDistribution(SingleFiniteDistribution):
     _argnames = ('p', 'succ', 'fail')
 
+    def __new__(cls, *args):
+        p = args[BernoulliDistribution._argnames.index('p')]
+        p_sym = sympify(p)
+
+        if fuzzy_not(fuzzy_and((p_sym.is_nonnegative, (p_sym - 1).is_nonpositive))):
+            raise ValueError("'p' must be: 0 <= p <= 1 . p = %s" % str(p))
```
Comment (smichr) [inline sympy/stats/frv_types.py]: or "p = %s is not in range [0, 1]."

File: sympy/stats/frv_types.py [hunk 2]
```diff
@@ -36,6 +36,13 @@ def dict(self):
 
     def __new__(cls, density):
         density = Dict(density)
+        for k in density.values():
+            k_sym = sympify(k)
+            if fuzzy_not(fuzzy_and((k_sym.is_nonnegative, (k_sym - 1).is_nonpositive))):
+                raise ValueError("Probability at a point must be between 0 and 1")
+        sum_sym = sum(density.values())
+        if sum_sym != 1:
+            raise ValueError("Total Probability must be equal to 1")
```
Comment (smichr) [inline sympy/stats/frv_types.py]: add a period after 1

**Comment by supreet11agrawal** [comment]
I have done the necessary changes. Kindly merge this PR

**Comment by oscarbenjamin** [comment]
The tests should be added before the PR is merged.

**Comment by supreet11agrawal** [comment]
Ok.Will do that

**Comment by supreet11agrawal** [comment]
@jksuom I have added the tests. However, as I am new to Sympy I am not sure if I have added them correctly or if I should add more. Tell me if a change is required

**Comment by jksuom** [comment]
Thanks, this looks good to me. The last test line is a bit longish but that can be fixed later.

**Pull request merged**